### ISSUE TITLE
`PGX_PG_SYS_GENERATE_BINDINGS_FOR_RELEASE=1 cargo pgx test` fails

### ISSUE THREAD

**Issue by eeeebbbbrrrr** [tcdi/pgx#835]
This is a fun thing from PR #830...

If the user trying to generate our bindings files runs, for example, in the `pgx-tests` crate:

```shell
$ PGX_PG_SYS_GENERATE_BINDINGS_FOR_RELEASE=1 cargo pgx test
```

**and** if their `$PGX_HOME/config.toml` file still has an entry for `pg10` (which is likely going to be true once #830 is released), then the pgx build fails with:

```
...
  --- stderr
  build_paths=BuildPaths { manifest_dir: "/Users/e_ridge/_work/pgx/pgx-pg-sys", out_dir: "/Users/e_ridge/_work/pgx/target/debug/build/pgx-pg-sys-c1be435560ad34c3/out", src_dir: "/Users/e_ridge/_work/pgx/pgx-pg-sys/src", shim_src: "/Users/e_ridge/_work/pgx/pgx-pg-sys/cshim", shim_dst: "/Users/e_ridge/_work/pgx/target/debug/build/pgx-pg-sys-c1be435560ad34c3/out/cshim" }
  Generating bindings for pg10
  Generating bindings for pg14
  Generating bindings for pg13
  Generating bindings for pg15
  Generating bindings for pg11
  Generating bindings for pg12
  thread '<unnamed>' panicked at 'Unable to generate bindings for pg10: NotExist("/Users/e_ridge/_work/pgx/pgx-pg-sys/include/pg10.h")', pgx-pg-sys/build.rs:590:29
...
```

So something in build.rs is trusting that whatever is in `$PGX_HOME/config.toml` matches its reality.  

I suppose the answer here is that we ignore any entries in there that we don't understand.  Or just not treat it as the source-of-truth for supported Postgres versions.  Or take the intersection of the two, perhaps.

**Comment by BradyBonnette** [comment]
Interesting, I tested this before I pushed it up. Weird! Ill check it out.

**Comment by thomcc** [comment]
This should only happen during local build so ignoring and reporting a `println!("cargo:warning=...")` seems appropriate.

**Comment by BradyBonnette** [comment]
@thomcc @eeeebbbbrrrr Do you feel it makes sense to have any unsupported versions rejected here?: https://github.com/tcdi/pgx/blob/master/pgx-pg-config/src/lib.rs#L372-L382 (that is, any unsupported version is just not returned as part of the iterator?)

Or would you rather see every version found in `$PGX_HOME/config.toml` be returned as part of the iterator (like it does now) and it's up to the caller to decide whether or not to use it?

**Comment by BradyBonnette** [comment]
Regarding my question above: It was decided to leave `pgx-pg-config` alone and let it do it's default behavior, which is to report all versions that are currently configured in `$PGX_HOME/config.toml` and leave it up to the caller/consumer to figure out what to do with that information. In this case, leave it up to build.rs to handle unsupported PG versions.

**Comment by BradyBonnette** [comment]
@eeeebbbbrrrr This issue can be closed as a result of #840 being merged
